fenix/app/src/main/java/org/mozilla/fenix/debugsettings/tabs/TabManager.kt

```diff
@@ -0,0 +1,226 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.debugsettings.tabs
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import androidx.compose.ui.unit.dp
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.ContentState
+import mozilla.components.browser.state.state.TabSessionState
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.lib.state.ext.observeAsState
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.Divider
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.ext.maxActiveTime
+import org.mozilla.fenix.tabstray.ext.isNormalTabInactive
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Tab manager UI for [DebugDrawer] that displays the tab counts and allows easy  bulk opening of tabs.
+ */
+@Composable
+fun TabManager(
+    store: BrowserStore,
+    inactiveTabsEnabled: Boolean,
```

Comment:
I don't think that we need to change anything here, but it strikes me as a little odd that we'd pass the `Store` and a parameter that we're presumably extracting in a layer above. Wondering whether a Composable should either be extracting every value itself or none

Reply:
so the thought process here is:
1. This is weird problem of using `BrowserStore`, which is an AC store and not a Fenix store, so it just contains the data for tabs in no UI-organizational manner, nor does it track the state of settings related to tabs.
2. The store is passed-in for just values right now, but it'll be needed to dispatch tab creation actions in a follow-up ticket.
3. I'm still thinking through how we're ultimately going to pass-in the drawer feature entries (and how that class/enum will be structured), and passing-in a single store seemed like the easy win for parameter simplification when those feature entries have to be initialized at runtime. This also ensures the observations only occur while the feature is open.
4. Yes, inactiveTabsEnabled, is obtained from settings, ultimately. We can't use `components.settings` at this level though, as it breaks previews, so we can't pass `Settings` as a dependency.

Thinking back on `TabsTray` and its parameter explosion, my goal for this screen is to have the root, feature-scoped composable node be the controlling data node, instead of having `DebugDrawer` or its caller be manager of the stuff specific to the tab manager/overview (e.g. tab stats and tab creation). 

Having a Store + an odd boolean is not ideal, but I can't think of an alternative besides:
- Creating a store/state for this sub feature
- Hoisting ALL of these observations to a level that may conflict with trying to keep the Debug Drawer ignorant of its contents.

Reply:
Another alternative would be creating an interstitial composable, so this top level handles the store + settings observations and then the `content` composable just gets raw data parameters.

Reply:
Thanks for the thoughts! I do have some responses, but they are strictly in the realm of theory and future considerations. Still happy with whatever we choose to do for now here as we will learn more as our goals change

> Thinking back on TabsTray and its parameter explosion

I would still like to find a way to avoid this if possible, and I'm wondering if something like a `TabManagerContract` should be expected by the top-level API. This removes any opinion about the backing architecture and data model, in the case that we want to expose it to consumers that aren't Fenix. Then, consumers can define their own way to scope their state to the contract

Something like:

```kotlin
// public API
data class TabManagerContract(
    val tabs: State<List<TabSessionState>>
    val inactiveTabs: State<..>
    val inactiveTabsEnabled: Boolean
    val addTab: (name: String) -> Unit
)

// Fenix API
fun toTabManagerContract(browserStore, settings): TabManagerContract

// if we wanted to get fancy and use context receivers instead
context(Settings)
fun BrowserStore.toTabManagerContract(): TabManagerContract

// or, more agnostic
fun BrowserStore.toTabManagerContract(inactiveTabsEnabled: Boolean): TabManagerContract
```

It's hard to imagine a world where anyone using the Debug Drawer isn't _also_ using the `BrowserStore`. However, something like `inactiveTabsEnabled` is more likely to live in an app-level Store, which we would not have access to if we were publishing this in A-C or otherwise. 

These thoughts also handily ignore touching on observability and smart recompositions 😅

Reply:
So the core assumption I've been formulating my thoughts around this has been: the debug drawer will be as framework agnostic as possible but the tab manager/tools is something specific to Fenix. So we have the following scenario:

- debug drawer is framework agnostic as possible 
- debug drawer sub features that have the potential to be used by other apps would follow the same
- debug drawer features that are Fenix specific can have our framework as dependencies

Reply:
Yeah, that makes sense to me!